homeassistant/components/yolink/sensor.py

```diff
@@ -94,26 +82,20 @@ def __init__(
         hass: HomeAssistant,
         coordinator: YoLinkCoordinator,
         description: YoLinkSensorEntityDescription,
-        device: dict,
-        client: YoLinkClient,
+        device: YoLinkDevice,
     ) -> None:
         """Init YoLink Sensor."""
-        super().__init__(hass, coordinator, device, client)
+        super().__init__(hass, coordinator, device)
         self.entity_description = description
-        self._attr_unique_id = f"{device[ATTR_DEVICE_ID]} {self.entity_description.key}"
-        self._attr_name = f"{device[ATTR_DEVICE_NAME]} ({self.entity_description.name})"
+        self._attr_unique_id = f"{device.device_id} {self.entity_description.key}"
+        self._attr_name = f"{device.device_name} ({self.entity_description.name})"
 
     @callback
     def update_entity_state(self, state: dict) -> None:
         """Update HA Entity State."""
         if state is None:
```

Comment:
This check doesn't match the type annotation of `state`.

Reply:
i will remove this check, state can't be None